docs/table/indexing.rst

```diff
@@ -0,0 +1,219 @@
+.. include:: references.txt
+.. |add_index| replace:: :func:`~astropy.table.Table.add_index`
+.. |index_mode| replace:: :func:`~astropy.table.Table.index_mode`
+
+Table indexing
+--------------
+
+Once a |Table| has been created, it is possible to create indexes on one or
+more columns of the table. An index internally sorts the rows of a table based
+on the index column(s), allowing for element retrieval by column value and
+improved performance for certain table operations.
+
+Creating an index
+^^^^^^^^^^^^^^^^^
+
+To create an index on a table, use the |add_index| method::
+
+   >>> from astropy.table import Table
+   >>> t = Table([(1, 2, 3, 4), (5, 6, 7, 8)], names=('a', 'b'))
```

Comment:
This first example should highlight how things get sorted, and the possibility of non-unique keys for column 'a'.  So use something like:
```
>>> t = Table([(2, 3, 2, 1), (8, 7, 6, 5)], names=('a', 'b'))```